src/renderer/components/tooltip/withTooltip.tsx

```diff
@@ -5,13 +5,14 @@
 
 // Tooltip decorator for simple composition with other components
 
-import type { HTMLAttributes, ReactNode } from "react";
-import React from "react";
+import type { ReactNode } from "react";
+import React, { useState } from "react";
```

Comment:
This lures business-logic in UI-code. Resist? :)

Reply:
I don't see how it does in this case, I am merely converting from class based components to function based ones. Yeah we want to remove the non-determinism to help with testing but it is going to be tricky (just tried quickly now) so I think we should leave until the future.

Reply:
It isn't really business logic to have unique IDs for tooltips. It is more of an implementation detail. Sure it is non-deterministic, but we can provide an ID for when we need it to be deterministic.